daphne_worker/src/dap.rs

```diff
@@ -297,12 +356,27 @@ impl<D> DapLeader<BearerToken> for DaphneWorkerConfig<D> {
             .get_task_config_for(&collect_req.task_id)
             .ok_or_else(|| DapError::fatal(INT_ERR_UNRECOGNIZED_TASK))?;
 
```

Comment:
Sampling the clock twice can lead to strange errors. Sample it once, then use it twice
```rust
let now = now();
if some_condition_that_depends_on_the_current_time(now) {
 ....
}
```

Reply:
I totally missed this. Thanks for pointing this out.